apis/hive/v1/hiveconfig_types.go

```diff
@@ -192,6 +192,17 @@ type AWSPrivateLinkConfig struct {
 	// +kubebuilder:default=Alias
 	// +optional
 	DNSRecordType AWSPrivateLinkDNSRecordType `json:"dnsRecordType,omitempty"`
+
+	// SharedHostedZoneDomains is a list of domain names for private hosted
+	// zones that are to be shared by multiple clusters' privatelink
+	// records. They must exist in the same account as the Hive cluster and
+	// be associated with the privatelink VPC. When a cluster's domain is a
```

Comment:
Right. But the wording of this docstring implies that it's the user's responsibility to ensure that the PHZs (or the domain names?)

> exist in the same account as the Hive cluster and [are] associated with the privatelink VPC.

Whereas in fact the controller is doing the creating and associating.

Maybe it would help me to understand what the user has to prepare in their account in order to be able to add a domain to this list.